# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/openat.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Some package start to "depend" on openat, ...

  - Rene Rebe <rene@exactcode.de>

--- /dev/null	2010-10-27 14:43:31.488782000 +0200
+++ dietlibc/syscalls.s/openat.S	2010-11-13 12:49:56.000000000 +0100
@@ -0,0 +1,3 @@
+#include "syscalls.h"
+
+syscall_weak(openat,openat,__libc_openat)
--- /dev/null	2010-10-27 14:43:31.488782000 +0200
+++ dietlibc/lib/openat64.c	2010-11-13 12:54:09.000000000 +0100
@@ -0,0 +1,13 @@
+#include <sys/types.h>
+#include <fcntl.h>
+
+#ifndef O_LARGEFILE
+#define O_LARGEFILE 0
+#endif
+
+int __libc_openat64(int dir, const char* file,int oflag,int mode);
+int __libc_openat64(int dir, const char* file,int oflag,int mode) {
+  return openat(dir,file,oflag|O_LARGEFILE,mode);
+}
+
+int openat64(int dir, const char* file,int oflag,...) __attribute__((weak,alias("__libc_openat64")));
